libcxx/include/string

```diff
@@ -754,13 +755,20 @@ public:
   // external memory. In such cases, the destructor is responsible for unpoisoning
   // the memory to avoid triggering false positives.
   // Therefore it's crucial to ensure the destructor is called.
+  //
+  // However, it is replaceable since implementing move-assignment as a destroy + move-construct
+  // will maintain the right ASAN state.
```

Comment:
This seems incorrect. The old annotations can very well be wrong for the new string.

Reply:
`__is_replaceable` means that you can destroy + move construct instead of assigning. This is orthogonal to whether performing those steps is equivalent to `memcpy` (which in the case of an ASAN-ified `string`, it isn't).

Reply:
Right, you're correct.